src/dmd/doc.d

```diff
@@ -1823,6 +1837,25 @@ extern (C++) bool isDitto(const(char)* comment)
     return false;
 }
 
+/*****************************************
+ * Return true if the given character is punctuation.
+ */
+private bool isPunctuation(const(char) c)
+{
+    static immutable bool[char.max+1] asciiPunctuation =
+       ['!': true, '"': true, '#': true,
+        '$': true, '%': true, '&': true, '\'': true, '(': true, ')': true,
+        '*': true, '+': true, ',': true, '-': true, '.': true, '/': true,
+        ':': true, ';': true, '<': true, '=': true, '>': true, '?': true,
+        '@': true, '[': true, '\\': true, ']': true, '^': true, '_': true,
+        '`': true, '{': true, '|': true, '}': true, '~': true];
+    return asciiPunctuation[c];
```

Comment:
If the given character is not in the associative array an error will be thrown.

Reply:
Except this is not an associative array, it's a regular array of length 256 as suggested by @BBasile. So no danger.

Reply:
Is this reinventing core.stdc.ctype.ispunct?

http://www.cplusplus.com/reference/cctype/

Reply:
After testing it, yes. Yes it is.